math/binary_exponent.cpp

```diff
@@ -19,29 +20,75 @@
  * \f}
  * Hence to calculate 2^10 we only need to multiply \f$2^8\f$ and \f$2^2\f$
  * skipping \f$2^1\f$ and \f$2^4\f$.
+ *
+ * @author [Mann Mehta](https://github.com/mann2108)
  */
 
-#include <iostream>
+#include <cassert>   /// for assert
+#include <iostream>  /// for IO operations
 
-/// Recursive function to calculate exponent in \f$O(\log(n))\f$ using binary
-/// exponent.
-int binExpo(int a, int b) {
+/**
+ * @brief Mathematical algorithms
+ * @namespace
+ */
+namespace math {
+/**
+ * @brief Recursive function to calculate exponent in \f$O(\log(n))\f$ using
+ * binary exponent.
+ * @param a base number
+ * @param b exponent number
+ * @return 1 if the exponent number is 0 the return value will always be 1
+ * @return 0 if the exponential number is < 0 ( 0 representing NULL since
+ * < 0 is an invalid exponential to use )
+ * @return (res * res * a) if modulo 2 of exponential is 0
+ * @return (res * res) if modulo 2 of exponential is 1
+ */
+uint64_t binExpo_recursive(int64_t a, int64_t b) {
```

Comment:
I would say to keep the change in type casting to a minimum - since you are changing the return type to use the `uint64_t` type then make your arguments `uint64_t` type too. The added benefit is that you don't need to add another check in your code about the valid exponent `if (b < 0)` and it would be less prone to type casting issues.

A general advice for C/C++ (it doesn't really do much to this small scale program but should still be relevant) - type casting should be avoided until you really need it and you know what you're doing...